doc/modules/decomposition.rst

```diff
@@ -278,6 +257,88 @@ factorization, while larger values shrink many coefficients to zero.
      R. Jenatton, G. Obozinski, F. Bach, 2009
 
 
+.. _kernel_PCA:
+
+Kernel Principal Component Analysis (kPCA)
+==========================================
+
+Exact Kernel PCA
+----------------
+
+:class:`KernelPCA` is an extension of PCA which achieves non-linear
+dimensionality reduction through the use of kernels (see :ref:`metrics`). It
+has many applications including denoising, compression and structured
+prediction (kernel dependency estimation). :class:`KernelPCA` supports both
+``transform`` and ``inverse_transform``.
+
+.. figure:: ../auto_examples/decomposition/images/sphx_glr_plot_kernel_pca_001.png
+    :target: ../auto_examples/decomposition/plot_kernel_pca.html
+    :align: center
+    :scale: 75%
+
+.. topic:: Examples:
+
+    * :ref:`sphx_glr_auto_examples_decomposition_plot_kernel_pca.py`
+
+.. _Randomized_kPCA:
+
+Kernel PCA using randomized SVD
+-------------------------------
+
+In :ref:`kernel_PCA`, the number of components found is equal to the number of
```

Comment:
```suggestion
In :class:`KernelPCA`, the number of components found is equal to the number of
```
Otherwise you're just linking to the section above

Reply:
Indeed, good catch